src/impl/icetransport.cpp

```diff
@@ -210,9 +215,17 @@ bool IceTransport::addRemoteCandidate(const Candidate &candidate) {
 	return juice_add_remote_candidate(mAgent.get(), string(candidate).c_str()) >= 0;
 }
 
-void IceTransport::gatherLocalCandidates(string mid) {
+void IceTransport::gatherLocalCandidates(string mid, std::vector<IceServer> additionalIceServers) {
 	mMid = std::move(mid);
 
+	int k = 0;
+	for (auto &server : additionalIceServers) {
+		addTURNServer(server);
+		if (++k >= MAX_TURN_SERVERS_COUNT) {
+			break;
+		}
+	}
```

Comment:
This mechanism will silently ignore any STUN server added in `additionalIceServers`. Is it by design?

Reply:
I will go fix that. I was only thinking about TURN servers when I implemented, will implement and test

Reply:
@paullouisageneau I will need to add another API to libjuice for that.

I haven't had anyone request this/I don't need it myself. Would it be better for libdatachannel if I implemented it though?

Personally it feels like more LoC/more bugs for a feature that I don't need yet! I am happy to do it though, your call!

Reply:
No need to implement it for now, but rather than silently ignoring the issue, you could rename the method `addIceServer()` and display a warning like "only TURN servers are supported as additional ICE servers".

Reply:
Done, we now log `Only TURN servers are supported as additional ICE servers`